docs/rules/no-unused-expressions.md

```diff
@@ -23,7 +23,9 @@ This rule, in its default state, does not require any arguments. If you would li
 * `allowShortCircuit` set to `true` will allow you to use short circuit evaluations in your expressions (Default: `false`).
 * `allowTernary` set to `true` will enable you use ternary operators in your expressions similarly to short circuit evaluations (Default: `false`).
 
-By default the following patterns are considered problems:
+These options allow unused expressions *only if all* of the code paths have *side effects*.
```

Comment:
I'm not sure `side effects` are clear in this case. Side effects were not mentioned prior to this line, so it might be unclear what does that mean.